File: 1122 relative-sort-array/solution.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int[] arr = new int[1001];
        for(int i:arr1) arr[i]++;
        int k =0;
        for(int i:arr2) {
            while(arr[i]-- > 0) {
                arr1[k++] = i;
            }
        }
        for(int i=0;i<arr.length;i++) {
            while(arr[i]-- > 0) {
                arr1[k++] = i;
            }
        }
        return arr1; 
    }
}

File: 1122 relative-sort-array/solution.go
func relativeSortArray(arr1 []int, arr2 []int) []int {
    freq:=make([]int,1001)
    for _,v:=range arr1 {
        freq[v]++
    }
    k:=0
    for _,v := range arr2 {
        for freq[v] > 0 {
            arr1[k] = v
            k++
            freq[v]--
        }
    }
    for i := 0; i < len(freq); i++ {
        for freq[i] > 0 {
            arr1[k] = i
            k++
            freq[i]--
        }
    }
    return arr1
}
